test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/controllers/BusinessListCYAControllerSpec.scala

```diff
@@ -107,6 +109,106 @@ class BusinessListCYAControllerSpec extends ControllerBaseSpec with MockIncomeTa
 
   }
 
+  "submit" should {
+    "return 303, SEE_OTHER" when {
+      "the connector returns successful json and submit with Yes option redirect to Initialse controller" in {
+        mockAuthSuccess()
+        mockGetSelfEmployments[Seq[SelfEmploymentData]]("Businesses")(Right(Some(Seq(
+          businessData
+        ))))
+
+        val result: Future[Result] = TestBusinessListCYAController.submit()(FakeRequest()
+          .withFormUrlEncodedBody(AddAnotherBusinessForm.addAnotherBusiness -> YesNoMapping.option_yes))
+
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.InitialiseController.initialise().url)
+
+      }
+
+      "the connector returns successful json and submit with No option redirect to Initialse controller" in {
+        mockAuthSuccess()
+        mockGetSelfEmployments[Seq[SelfEmploymentData]]("Businesses")(Right(Some(Seq(
+          businessData
+        ))))
+
+        val result: Future[Result] = TestBusinessListCYAController.submit()(FakeRequest()
+          .withFormUrlEncodedBody(AddAnotherBusinessForm.addAnotherBusiness -> YesNoMapping.option_no))
+
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.BusinessAccountingMethodController.show().url)
+      }
+
+    }
+
+    "return (303) redirect to business start date page" when {
+      "no businesses are returned" in {
+        mockAuthSuccess()
+        mockGetSelfEmployments[Seq[SelfEmploymentData]]("Businesses")(Right(None))
+
+        val result: Future[Result] = TestBusinessListCYAController.submit()(FakeRequest()
+          .withFormUrlEncodedBody(AddAnotherBusinessForm.addAnotherBusiness -> YesNoMapping.option_yes))
```

Comment:
This test isn't really meaningful unless you set the option to "No". It has no behaviour difference to selecting "Yes" in any circumstance.

Reply:
That's correct, I was only looking at empty getselfemployments. 
Done.